### PR TITLE
GPT-J specific half precision on CPU note

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR adds a note to the GPT-J model doc indicating that the half precision on CPU example is specific to the model, and doesn't generally apply. Related to https://github.com/huggingface/transformers/issues/21989

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
I'm really not sure this model actually runs on CPU since it does contain layernorms. I was thinking more of putting a CUDA device in the example.

**Comment by MKhalusova** [comment]
My bad, I misunderstood the interaction in the issue. Tried the example, and it does indeed not work. I'll do another pass.

File: docs/source/en/model_doc/gptj.mdx [hunk 1]
```diff
@@ -85,7 +84,8 @@ model.
 >>> from transformers import GPTJForCausalLM, AutoTokenizer
 >>> import torch
 
->>> model = GPTJForCausalLM.from_pretrained("EleutherAI/gpt-j-6B", torch_dtype=torch.float16)
+>>> device = "cuda" if torch.cuda.is_available() else "cpu"
```
Comment (sgugger) [inline docs/source/en/model_doc/gptj.mdx:87]: Here we should always use `"cuda"` since there will be an error on CPU anyway.

**Review by sgugger — APPROVED** [review]
Thnaks a lot!

**Pull request merged**